### PR TITLE
Add rdkitPath and rdkitWorkerPath props

### PR DESCRIPTION
**Pull request by atfera**
Added a custom worker path props ~and revert the change for initialRdkitInstance in v2~

EDIT:
It seems we can't pass an rdkit instance in postmessage worker (Can't be cloned). So I removed this change and this PR will only focus on adding a custom rdkit worker path props.

EDIT 2:
Added custom rdkitPath props

### CODE REVIEW

File: src/contexts/RDKitContext.tsx [hunk 1]
```diff
@@ -10,26 +11,34 @@ export interface RDKitContextValue {
...
 
     const initialise = async () => {
-      workerInstance = initWorker();
+      workerInstance = initWorker(rdkitWorkerPath);
       // await rdkit module init in worker before starting using the worker
       await postWorkerJob(workerInstance, {
         actionType: RDKIT_WORKER_ACTIONS.INIT_RDKIT_MODULE,
         key: 'worker-init',
-        payload: { cache, preferCoordgen },
+        payload: { cache, preferCoordgen, initialRdkitInstance },
```
Comment (RamziWeslati) [inline src/contexts/RDKitContext.tsx:41]: I thought I couldn't pass an `rdkitModule` from main thread to the worker thread, as it failed to serialize it, did this work for you? passing around `rdkitModule` from main to worker
@atfera

Comment (atfera) [reply src/contexts/RDKitContext.tsx]: You are right, cant be cloned. I removed it.

**Review by RamziWeslati — APPROVED** [review]

File: package-lock.json
Comment (RamziWeslati) [inline package-lock.json:1]: why was this removed?

File: README.md [hunk 2]
```diff
@@ -21,21 +25,24 @@ import { RDKitProvider } from '@iktos/rdkit-provider';
...
+| preferCoordgen         | `boolean`                                                                                 | will be passed to [@rdkit/rdkitjs prefer_coordgen](https://docs.rdkitjs.com/interfaces/RDKitModule.html#prefer_coordgen.prefer_coordgen-1) to use Schrodinger’s open-source Coordgen library to generate 2D coordinates of molecules | optional          |
+| removeHs               | `boolean`                                                                                 | toggles removing hydrogens molecules. Defaults to true                                                                                                                                                                               | optional          |
+| `initialRdkitInstance` | `RDKitModule` from `@rdkit/rdkitjs`                                                       | pass an instance of to expose by the context `RDKitModule`, if not passed rdkit-provider creates one for you                                                                                                                         | optional          |
```
Comment (RamziWeslati) [inline README.md:43]: I think`initialRdkitInstance` is no longer relevant

File: src/worker/utils/initRDKit.ts [hunk 3]
```diff
@@ -25,16 +25,22 @@
...
+  const path = rdkitPath || '/RDKit_minimal.js';
+  const url = new URL(path, globalThis.origin);
   //@ts-ignore
-  importScripts(`${globalThis.origin}/RDKit_minimal.js`);
+  importScripts(url);
   if (!globalThis.initRDKitModule) return;
   await globalThis.initRDKitModule().then((rdkitModule) => {
     globalThis.workerRDKit = rdkitModule;
   });
+  //@ts-ignore
```
Comment (RamziWeslati) [inline src/worker/utils/initRDKit.ts:43]: What was wrong with ts here?

File: README.md (hunk 2 above)
Comment (atfera) [reply README.md]: Should be `initialWorkerInstance`, not rdkit module.
Changed it.
I added this props so we can pass an rdkit worker instance. It the same problem with previous version and rdkit module, I need a way to call the rdkit worker  outside react context.

File: package-lock.json
Comment (atfera) [reply package-lock.json:1]: Hmm strange, maybe because I tried to change the package version for a pre-release, but I don't know why it changed the dependencies.

File: src/worker/utils/initRDKit.ts (hunk 3 above)
Comment (atfera) [reply src/worker/utils/initRDKit.ts]: TS error: `prefer_coordgen` not existing on type `never`.
Changed the function to avoid it.

**Review by RamziWeslati — APPROVED** [review]
